### PR TITLE
Merge the benchmarking script

### PR DESCRIPTION
**Pull request by rsmmr**
The script isn't perfect yet, but will do the job for now. No need to review in detail.

I added documentation on how to use it to the Developer's manual. I also prepared a preprocessed archive with test data, generated from the Zeek external test suite traces. Need to upload that somewhere still, it's pretty large.

This PR also includes fixes for a couple of regressions (functionality & performance). We remain a bit slower still than Zeek (we used to be closer), but nothing obviously broken sticks out in the perf output at least.

### CODE REVIEW

File: spicy/runtime/src/driver.cc [hunk 1]
```diff
@@ -21,6 +21,14 @@ using hilti::rt::fmt;
 
 using namespace spicy::rt;
 
+#ifndef NDEBUG
+#define DRIVER_DEBUG(...) debug(__VA_ARGS__)
```
Comment (bbannier) [inline spicy/runtime/src/driver.cc:25]: Won't this break if we ever have to call e.g., `this->debug(...)`? Maybe we could make the forwarding in our `debug` (the non-`virtual` one) conditional on `NDEBUG` so it becomes a no-op for release code.

File: hilti/runtime/include/fiber.h [hunk 2]
```diff
@@ -68,6 +68,8 @@ class Fiber {
     void resume();
     void abort();
 
+    bool isDone() { return _state != State::Running && _state != State::Yielded; }
```
[the remark this answers is not in the document]
Comment (rsmmr) [reply hilti/runtime/include/fiber.h:71]: Different class, Fiber doesn't have a _done. I'll do the switch.

File: hilti/runtime/include/fiber.h [hunk 3]
```diff
@@ -213,6 +219,7 @@ class Resumable {
     }
 
     std::unique_ptr<detail::Fiber> _fiber;
+    bool _done = false;
```
[the remark this answers is not in the document]
Comment (rsmmr) [reply hilti/runtime/include/fiber.h:222]: I think we need the "regular termination" too, or at least it's easier that way.

File: spicy/runtime/src/driver.cc [hunk 4]
```diff
@@ -236,7 +236,7 @@ driver::ParsingState::State driver::ParsingState::_process(size_t size, const ch
                 input->freeze();
 
                 _resumable = _parser->parse1(input, {});
-                if ( ! _resumable )
+                if ( ! *_resumable )
```
[the remark this answers is not in the document]
Comment (rsmmr) [reply spicy/runtime/src/driver.cc:236]: Not sure what to test?

File: spicy/runtime/src/driver.cc (hunk 1 above)
Comment (rsmmr) [reply spicy/runtime/src/driver.cc:25]: It has to be done before we call any debug(), and the virtual is also called directly.

If anything, I'd do two macros, but it's also just an implementation detail of this file at this point, not sure it's worth much more. If it breaks, we'll notice.

**Review by bbannier — APPROVED** [review]

**Review by bbannier — APPROVED** [review]

**Pull request merged**